launch/test/launch/substitutions/test_boolean_substitution.py

```diff
@@ -54,3 +56,39 @@ def test_or_substitution():
         OrSubstitution('not-condition-expression', 'true').perform(lc)
     with pytest.raises(SubstitutionFailure):
         OrSubstitution('true', 'not-condition-expression').perform(lc)
+
+
+def test_any_substitution():
+    lc = LaunchContext()
+    assert AnySubstitution().perform(lc) == 'false'
+    assert AnySubstitution('true').perform(lc) == 'true'
+    assert AnySubstitution('false').perform(lc) == 'false'
+    assert AnySubstitution('true', 'true').perform(lc) == 'true'
+    assert AnySubstitution('true', 'false').perform(lc) == 'true'
+    assert AnySubstitution('false', 'true').perform(lc) == 'true'
+    assert AnySubstitution('false', 'false').perform(lc) == 'false'
+    assert AnySubstitution('true', 'true', 'true').perform(lc) == 'true'
+    assert AnySubstitution('true', 'true', 'false').perform(lc) == 'true'
+    assert AnySubstitution('false', 'false', 'false').perform(lc) == 'false'
```

Comment:
If we support `'1'` as a truthy value in other substitutions, then I would expect the following to be true as well:

```
AnySubstitution('false', 'false', '1').perform(lc) == 'true'
```

Again, I'm not sure if we should support this, but we should at least be consistent.

Reply:
Considering that we suppor the strings `'1'` and `'0'` for conditions, I think we should add support for them to the any and all substitutions.

Reply:
Will do!

Reply:
They are now implicitly supported following the merge of #651. Tests have been implemented to show that this is the case
https://github.com/ros2/launch/pull/649/commits/4de282662e85d5ded9886d87ff36d5bd6225b6a4